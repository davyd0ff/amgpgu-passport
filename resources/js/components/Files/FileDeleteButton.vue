<template>
  <button class="btn btn-danger" v-on:click="onClick">
    <i class="medium material-icons">delete</i>
  </button>
</template>

<script>
export default {
  name: 'FileDeleteButton',
  methods: {
    onClick: function () {
      this.$emit('delete-file');
      // this.onDeleteFile().then(() => {
      //   this.$message('DELETE_FILE_FILE_WAS_DELETED');
      // });
      // .catch((_) => {});
    },
  },
};
</script>

<style scoped></style>
