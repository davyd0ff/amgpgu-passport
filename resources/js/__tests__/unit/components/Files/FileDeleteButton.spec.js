import { shallowMount, createLocalVue } from '@vue/test-utils';
import FileDeleteButton from '@/components/Files/FileDeleteButton.vue';
import { ExplorerSync } from 'prettier/third-party';

describe('FileDeleteButton.vue', () => {
  it('render', () => {
    const wrapper = shallowMount(FileDeleteButton);

    expect(wrapper).toMatchSnapshot();
  });

  describe('press on button', () => {
    const $message = jest.fn();
    const makeWrapper = (options) =>
      shallowMount(FileDeleteButton, {
        ...options,
        mocks: { $message },
      });

    it('file was deleted', async () => {
      $message.mockReset();
      const onDeleteFile = jest.fn(() => Promise.resolve({}));
      const wrapper = makeWrapper({
        propsData: { onDeleteFile },
      });

      await wrapper.find('button').trigger('click');

      expect(onDeleteFile).toHaveBeenCalled();
      expect($message).toHaveBeenCalledWith('DELETE_FILE_FILE_WAS_DELETED');
    });

    // it('file was not deleted', () => {
    //   const $message = jest.fn();
    //   const onDeleteFile = jest.fn(() => Promise.reject({}));
    //   const wrapper = makeWrapper({
    //     propsData: { onDeleteFile },
    //     mocks: { $message },
    //   });

    //   wrapper.find('button').trigger('click');

    //   expect(onDeleteFile).toHaveBeenCalled();
    //   expect($message).not.toHaveBeenCalled();
    // });
  });
});
