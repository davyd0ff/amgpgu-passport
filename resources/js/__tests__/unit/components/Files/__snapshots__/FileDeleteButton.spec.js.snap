// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDeleteButton.vue render 1`] = `<button class="btn btn-danger"><i class="medium material-icons">delete</i></button>`;
